ui/pages/send/send-content/send-hex-data-row/send-hex-data-row.js

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from 'react';
+import { isHexString } from 'ethereumjs-util';
+import { useDispatch, useSelector } from 'react-redux';
+import { getSendHexData, updateSendHexData } from '../../../../ducks/send';
+import SendRowWrapper from '../send-row-wrapper';
+import Dialog from '../../../../components/ui/dialog';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+
+const SendHexDataRow = () => {
+  const [shouldShowError, setShouldShowError] = useState(false);
+
+  const t = useI18nContext();
+
+  const data = useSelector(getSendHexData);
+
+  const dispatch = useDispatch();
+
+  const onInput = (event) => {
+    const isValidHex = isHexString(event.target.value) || !event.target.value;
+    setShouldShowError(!isValidHex);
+    const hexData = event.target.value.replace(/\n/gu, '') || null;
+    dispatch(updateSendHexData(hexData));
+  };
+
+  const renderError = (error) => {
+    return (
+      <Dialog
+        type="error"
+        className="send__error-dialog"
+        data-testid="hex-data-error-message"
+      >
+        {t(error)}
+      </Dialog>
+    );
+  };
```

Comment:
This pattern is used elsewhere but lets push ourselves towards the strict mode messages system by not passing variables to the t function.

```suggestion
  const renderError = (error) => {
    let errorMessage = '';
    switch (error) {
       case INVALID_HEX_INPUT: 
         errorMessage = t('INVALID_HEX_INPUT');
         break;
      case OTHER_POSSIBLE_ERROR:
        // repeat
      default:
        // sensible default
     }
    return (
      <Dialog
        type="error"
        className="send__error-dialog"
        data-testid="hex-data-error-message"
      >
        {errorMessage}
      </Dialog>
    );
  };
```

Reply:
@brad-decker thinking about this, this component is ever going to display hex data errors so we can do this instead

```suggestion
  const renderError = () => {
    return (
      <Dialog
        type="error"
        className="send__error-dialog"
        data-testid="hex-data-error-message"
      >
        {t('invalidHexString')}
      </Dialog>
    );
  };
```